contracts/SimpleAssetsExchange.sol

```diff
@@ -33,24 +54,23 @@ contract SimpleAssetsExchange is Ownable, IAssetsExchange {
     emit PriceProviderChanged(address(priceProvider));
   }
 
-
   /* ========== MUTATIVE FUNCTIONS ========== */
 
 
   /**
-   * Buys selected asset with AVAX
+   * Buys selected asset with AVAX using the Pangolin DEX
+   * Refunds unused AVAX to the msg.sender
    * @dev _asset asset code
    * @dev _amount amount to be bought
   **/
-  function buyAsset(bytes32 _asset, uint256 _amount) payable override external {
+  function buyAsset(bytes32 _asset, uint256 _amount) payable override external RefundRemainder{
     uint256 amountIn = _amount * priceProvider.getPrice(_asset) / 1 ether;
     require(amountIn > 0, "Incorrect input amount");
     require(msg.value >= amountIn, "Not enough funds provided");
 
+    pangolinRouter.swapAVAXForExactTokens{ value: msg.value }(_amount, getPathForAVAXtoToken(_asset), address(this), block.timestamp + 100);
```

Comment:
What's the purpose of "block.timestamp+100" ?

Reply:
Nice catch. It was a leftover from JS scripts testing and is no longer required. Hence, I deleted it in the newest commit.